charts/library/common/templates/spawner/traefik/_middleware.tpl

```diff
@@ -12,13 +12,40 @@
     {{- $_ := set $.Values.ingressMiddlewares "traefik" dict -}}
   {{- end -}}
 
+
   {{- range $name, $middleware := $.Values.ingressMiddlewares.traefik -}}
 
+
+
     {{- $enabled := (include "tc.v1.common.lib.util.enabled" (dict
                     "rootCtx" $ "objectData" $middleware
                     "name" $name "caller" "Middleware"
                     "key" "middlewares")) -}}
 
+    {{- if ne $enabled "true" -}}
+      {{- range $ingressName, $ingress := $.Values.ingress }}
+        {{- if and $ingress.enabled $ingress.integrations  $ingress.integrations.traefik $ingress.integrations.traefik.enabled }}
```

Comment:
just notes for me when I go over it to polish/add tests:

- ingress.enabled might be a tpl.
- needs some whitespace cleanup (`-`)
- break loops on the first enabled=true

Reply:
> ingress.enabled might be a tpl.

Fixed that one